state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1057,6 +1057,17 @@ impl<T: Config> PoolApi for Pallet<T> {
 	fn sweep(who: &T::AccountId) -> DispatchResult {
 		Self::inner_sweep(who)
 	}
+
+	fn open_order_count(
```

Comment:
I find this highly unlikely and I'm not sure how we can avoid it without re-writing the types in the pools pallet. We could add an iter() method to the Pools type, but this would suffer from the same problem (no type safety).

Reply:
Hmm, maybe the solution isn't so much testing or type safety, but just changing the logic in `deregister_lp_account`. Why not just refund the amounts they have to their refund address instead of burning them?

Reply:
Whether we automatically send the funds to their refund address or not has nothing do do with open orders - they would still have to cancel these.

I feel like auto-withdrawals would add a bit more complexity that would need to be thought about and tested. I would rather not spend more time on this right now. The main point of this PR was defensive: ensuring users clean up after themselves. I'll log this as a future improvement in Linear.

Reply:
If an LP has funds in free balance, but no orders. Then using --force would mean that
```rust
			ensure!(
				Asset::all().filter(|asset| *asset != STABLE_ASSET).all(|asset| {
					T::PoolApi::open_order_count(&account_id, asset, STABLE_ASSET)
						.unwrap_or_default() == 0
				}),
				Error::<T>::OpenOrdersRemaining
			);
			ensure!(
				force || Self::asset_balances(&account_id)?.iter().all(|(_, amount)| *amount == 0),
				Error::<T>::FundsRemaining
			);
```
These two checks are true and we erase their balance. That's incredibly dangerous. At the very least we should remove the force and they should have to remove their balances manually.

Reply:
I disagree with 'incredibly dangerous'. It requires the LP to specifically opt-in and the idea was to allow them to deregister even when their free balance was too low to be worth withdrawing. But as discussed - I'll remove the `force` option here and for the broker.